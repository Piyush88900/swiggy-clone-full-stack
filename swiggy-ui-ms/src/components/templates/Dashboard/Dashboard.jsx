/**
 *
 * Dashboard
 *
 */
import React from 'react';
import { Box, Grid, Typography, Container } from '@material-ui/core';
import Layout from '../Layout';
import Link from '../../atoms/Link';

import GridComponent from '../../organisms/Grid';

const Dashboard = () => {
  return (
    <Layout>
      <Container>
        <Grid item xs={12}>
          <Box my={2}>
            <Link href="/">Home</Link>
          </Box>
        </Grid>
        <Grid item xs={12}>
          <GridComponent></GridComponent>
        </Grid>
      </Container>
    </Layout>
  );
};

export default Dashboard;
