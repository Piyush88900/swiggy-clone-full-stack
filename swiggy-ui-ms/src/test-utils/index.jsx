import React from 'react';
import PropTypes from 'prop-types';
import { render } from '@testing-library/react';
import { MockedProvider } from '@apollo/client/testing';
import { ThemeProvider } from '@material-ui/core/styles';
import { LabelProvider } from '../stores/labels';
import { theme } from '../theme';

const AllTheProviders = ({ children, mocks = [] }) => {
  return (
    <MockedProvider mocks={mocks} addTypename={false}>
      <ThemeProvider theme={theme}>
        <LabelProvider>{children}</LabelProvider>
      </ThemeProvider>
    </MockedProvider>
  );
};

AllTheProviders.propTypes = {
  children: PropTypes.node.isRequired,
  mocks: PropTypes.arrayOf(PropTypes.shape({})),
};

AllTheProviders.defaultProps = {
  mocks: [],
};

const customRender = (ui, options) => {
  const { wrapperProps = {}, ...customOptions } = options || {};
  return render(ui, {
    wrapper: (props) => {
      return <AllTheProviders {...props} {...wrapperProps} />;
    },
    ...customOptions,
  });
};

// re-export everything
export * from '@testing-library/react';
export * from './withTestRouter';

// override render method
export { customRender as render };
