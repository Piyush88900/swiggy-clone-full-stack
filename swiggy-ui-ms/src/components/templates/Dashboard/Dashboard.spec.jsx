import React from 'react';
import { render, withTestRouter } from '../../../test-utils';

import Dashboard from '.';

describe('Dashboard', () => {
  it('should render correctly', () => {
    const { asFragment } = render(withTestRouter(<Dashboard />));
    expect(asFragment()).toMatchSnapshot();
  });
});
