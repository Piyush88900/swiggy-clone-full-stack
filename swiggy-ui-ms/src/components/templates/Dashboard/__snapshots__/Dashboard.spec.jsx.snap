// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard should render correctly 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-1"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular"
        >
          <h6
            class="MuiTypography-root makeStyles-title-3 MuiTypography-h6"
          >
            Hiring & Staffing
          </h6>
        </div>
      </div>
    </header>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          class="MuiBox-root MuiBox-root-4"
        >
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
            href="/"
          >
            Home
          </a>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          style="width: 100%; height: 100%;"
        >
          <div
            class="ag-theme-balham"
            id="myGrid"
            style="height: 500px; width: 100%;"
          >
            <div
              style="height: 100%;"
            >
              <div
                class="ag-root-wrapper ag-layout-normal ag-ltr"
              >
                
            
                <div
                  class="ag-root-wrapper-body ag-layout-normal"
                  ref="rootWrapperBody"
                >
                  
                
                  <!--AG-GRID-COMP-->
                  <div
                    class="ag-root ag-unselectable ag-layout-normal"
                    ref="gridPanel"
                    role="grid"
                    unselectable="on"
                  >
                    
        
                    <!--AG-HEADER-ROOT-->
                    <div
                      class="ag-header ag-pivot-on"
                      ref="headerRoot"
                      role="row"
                      style="height: 32px; min-height: 32px;"
                      unselectable="on"
                    >
                      
            
                      <div
                        class="ag-pinned-left-header ag-hidden"
                        ref="ePinnedLeftHeader"
                        role="presentation"
                        style="width: 0px; max-width: 0px; min-width: 0px;"
                      >
                        <div
                          class="ag-header-row"
                          role="presentation"
                          style="top: 0px; height: 32px; width: 0px;"
                        />
                      </div>
                      
            
                      <div
                        class="ag-header-viewport"
                        ref="eHeaderViewport"
                        role="presentation"
                      >
                        
                
                        <div
                          class="ag-header-container"
                          ref="eHeaderContainer"
                          role="presentation"
                          style="width: 0px;"
                        >
                          <div
                            class="ag-header-row"
                            role="presentation"
                            style="top: 0px; height: 32px; width: 0px;"
                          />
                        </div>
                        
            
                      </div>
                      
            
                      <div
                        class="ag-pinned-right-header ag-hidden"
                        ref="ePinnedRightHeader"
                        role="presentation"
                        style="width: 0px; max-width: 0px; min-width: 0px;"
                      >
                        <div
                          class="ag-header-row"
                          role="presentation"
                          style="top: 0px; height: 32px; width: 0px;"
                        />
                      </div>
                      
        
                    </div>
                    
        
                    <div
                      class="ag-floating-top"
                      ref="eTop"
                      role="presentation"
                      style="min-height: 0px; height: 0px; display: none; overflow-y: hidden;"
                      unselectable="on"
                    >
                      
            
                      <div
                        class="ag-pinned-left-floating-top ag-hidden"
                        ref="eLeftTop"
                        role="presentation"
                        unselectable="on"
                      />
                      
            
                      <div
                        class="ag-floating-top-viewport"
                        ref="eTopViewport"
                        role="presentation"
                        unselectable="on"
                      >
                        
                
                        <div
                          class="ag-floating-top-container"
                          ref="eTopContainer"
                          role="presentation"
                          style="width: 0px;"
                          unselectable="on"
                        />
                        
            
                      </div>
                      
            
                      <div
                        class="ag-pinned-right-floating-top ag-hidden"
                        ref="eRightTop"
                        role="presentation"
                        unselectable="on"
                      />
                      
            
                      <div
                        class="ag-floating-top-full-width-container ag-hidden"
                        ref="eTopFullWidthContainer"
                        role="presentation"
                        unselectable="on"
                      />
                      
        
                    </div>
                    
        
                    <div
                      class="ag-body-viewport ag-layout-normal ag-row-animation"
                      ref="eBodyViewport"
                      role="presentation"
                      unselectable="on"
                    >
                      
            
                      <div
                        class="ag-pinned-left-cols-container ag-hidden"
                        ref="eLeftContainer"
                        role="presentation"
                        style="height: 1px;"
                        unselectable="on"
                      />
                      
            
                      <div
                        class="ag-center-cols-clipper"
                        ref="eCenterColsClipper"
                        style="height: 1px;"
                      >
                        
                
                        <div
                          class="ag-center-cols-viewport"
                          ref="eCenterViewport"
                          role="presentation"
                          style="height: calc(100% + 0px);"
                          unselectable="on"
                        >
                          
                    
                          <div
                            class="ag-center-cols-container"
                            ref="eCenterContainer"
                            role="presentation"
                            style="width: 0px; height: 1px;"
                            unselectable="on"
                          />
                          
                
                        </div>
                        
            
                      </div>
                      
            
                      <div
                        class="ag-pinned-right-cols-container ag-hidden"
                        ref="eRightContainer"
                        role="presentation"
                        style="height: 1px;"
                        unselectable="on"
                      />
                      
            
                      <div
                        class="ag-full-width-container"
                        ref="eFullWidthContainer"
                        role="presentation"
                        style="height: 1px;"
                        unselectable="on"
                      />
                      
        
                    </div>
                    
        
                    <div
                      class="ag-floating-bottom"
                      ref="eBottom"
                      role="presentation"
                      style="min-height: 0px; height: 0px; display: none; overflow-y: hidden;"
                      unselectable="on"
                    >
                      
            
                      <div
                        class="ag-pinned-left-floating-bottom ag-hidden"
                        ref="eLeftBottom"
                        role="presentation"
                        unselectable="on"
                      />
                      
            
                      <div
                        class="ag-floating-bottom-viewport"
                        ref="eBottomViewport"
                        role="presentation"
                        unselectable="on"
                      >
                        
                
                        <div
                          class="ag-floating-bottom-container"
                          ref="eBottomContainer"
                          role="presentation"
                          style="width: 0px;"
                          unselectable="on"
                        />
                        
            
                      </div>
                      
            
                      <div
                        class="ag-pinned-right-floating-bottom ag-hidden"
                        ref="eRightBottom"
                        role="presentation"
                        unselectable="on"
                      />
                      
            
                      <div
                        class="ag-floating-bottom-full-width-container ag-hidden"
                        ref="eBottomFullWidthContainer"
                        role="presentation"
                        unselectable="on"
                      />
                      
        
                    </div>
                    
        
                    <div
                      class="ag-body-horizontal-scroll"
                      ref="eHorizontalScrollBody"
                      style="height: 0px; max-height: 0px; min-height: 0px;"
                    >
                      
            
                      <div
                        class="ag-horizontal-left-spacer ag-scroller-corner"
                        ref="eHorizontalLeftSpacer"
                        style="width: 0px; max-width: 0px; min-width: 0px;"
                      />
                      
            
                      <div
                        class="ag-body-horizontal-scroll-viewport"
                        ref="eBodyHorizontalScrollViewport"
                        style="height: 0px; max-height: 0px; min-height: 0px;"
                      >
                        
                
                        <div
                          class="ag-body-horizontal-scroll-container"
                          ref="eBodyHorizontalScrollContainer"
                          style="width: 0px; height: 0px; max-height: 0px; min-height: 0px;"
                        />
                        
            
                      </div>
                      
            
                      <div
                        class="ag-horizontal-right-spacer ag-scroller-corner"
                        ref="eHorizontalRightSpacer"
                        style="width: 0px; max-width: 0px; min-width: 0px;"
                      />
                      
        
                    </div>
                    
        
                    <!--AG-OVERLAY-WRAPPER-->
                    <div
                      class="ag-overlay ag-hidden"
                      ref="overlayWrapper"
                    >
                      
            
                      <div
                        class="ag-overlay-panel"
                        role="presentation"
                      >
                        
                
                        <div
                          class="ag-overlay-wrapper ag-layout-normal"
                          ref="eOverlayWrapper"
                        />
                        
            
                      </div>
                      
        
                    </div>
                    
    
                  </div>
                  
            
                </div>
                
            
                <!--AG-PAGINATION-->
                <div
                  class="ag-paging-panel ag-unselectable"
                >
                  
                
                  <span
                    class="ag-paging-row-summary-panel"
                    ref="eSummaryPanel"
                  >
                    
                    
                    <span
                      ref="lbFirstRowOnPage"
                    >
                      0
                    </span>
                     to 
                    <span
                      ref="lbLastRowOnPage"
                    >
                      0
                    </span>
                     of 
                    <span
                      ref="lbRecordCount"
                    >
                      0
                    </span>
                    
                
                  </span>
                  
                
                  <span
                    class="ag-paging-page-summary-panel"
                  >
                    
                    
                    <button
                      class="ag-paging-button"
                      disabled=""
                      ref="btFirst"
                      type="button"
                    >
                      First
                    </button>
                    
                    
                    <button
                      class="ag-paging-button"
                      disabled=""
                      ref="btPrevious"
                      type="button"
                    >
                      Previous
                    </button>
                    
                    Page 
                    <span
                      ref="lbCurrent"
                    >
                      0
                    </span>
                     of 
                    <span
                      ref="lbTotal"
                    >
                      0
                    </span>
                    
                    
                    <button
                      class="ag-paging-button"
                      disabled=""
                      ref="btNext"
                      type="button"
                    >
                      Next
                    </button>
                    
                    
                    <button
                      class="ag-paging-button"
                      disabled=""
                      ref="btLast"
                      type="button"
                    >
                      Last
                    </button>
                    
                
                  </span>
                  
            
                </div>
                
        
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
